custom_components/evohome_cc/__init__.py

```diff
@@ -158,8 +195,108 @@ async def load_system_config(store) -> Optional[Dict]:
         broker.update, hass_config[DOMAIN][CONF_SCAN_INTERVAL]
     )
 
+    setup_service_functions(hass, broker)
+
     return True
 
+@callback
+def setup_service_functions(hass: HomeAssistantType, broker):
+    @verify_domain_control(hass, DOMAIN)
+    async def force_refresh(call) -> None:
+        """Obtain the latest state data via the vendor's RESTful API."""
+        await broker.update()
+
+    @verify_domain_control(hass, DOMAIN)
+    async def set_system_mode(call) -> None:
+        """Set the system mode."""
+        payload = {
+            CONF_UNIQUE_ID: broker.client.evo.unique_id,
+            "service": call.service,
+            "data": call.data,
+        }
+        async_dispatcher_send(hass, DOMAIN, payload)
+
+    @verify_domain_control(hass, DOMAIN)
+    async def set_zone_override(call) -> None:
+        """Set the zone override (setpoint)."""
+
+        entity_id = call.data[ATTR_ENTITY_ID]
+
+        registry = await hass.helpers.entity_registry.async_get_registry()
+        registry_entry = registry.async_get(entity_id)
+
+        if registry_entry is None or registry_entry.platform != DOMAIN:
+            raise ValueError(f"'{entity_id}' is not a known {DOMAIN} entity")
+
+        if registry_entry.domain != "climate":
+            raise ValueError(f"'{entity_id}' is not an {DOMAIN} controller/zone")
+
+        payload = {
+            CONF_UNIQUE_ID: registry_entry.unique_id,
+            "service": call.service,
+            "data": call.data,
+        }
+
+        async_dispatcher_send(hass, DOMAIN, payload)
+
+    hass.services.async_register(DOMAIN, SVC_REFRESH_SYSTEM, force_refresh)
+
+    system_mode_schemas = []
+
+    # Not all systems support "AutoWithReset": register this handler only if required
+    if [m for m in SYSTEM_MODE_LOOKUP if m == EVO_MODE_RESET]:
```

Comment:
I think you've copied this from the **evohome** integration without understanding the nuances?